packages/editor-ui/src/components/MainSidebar.vue

```diff
@@ -431,7 +431,14 @@ export default mixins(
 						const importConfirm = await this.confirmMessage(`When you switch workflows your current workflow changes will be lost.`, 'Save your Changes?', 'warning', 'Yes, switch workflows and forget changes');
 						if (importConfirm === true) {
 							this.$store.commit('setStateDirty', false);
-							this.$router.push({ name: 'NodeViewNew' });
+							if (this.$router.currentRoute.name === 'NodeViewNew') {
```

Comment:
this approach does not seem to work if you navigate to `New` multiple times in the same session... and it's not pretty
can you try to find a more fundamental fix?
<img width="1440" alt="Screen Shot 2021-06-01 at 2 01 42 PM" src="https://user-images.githubusercontent.com/4711238/120312933-065ba980-c2e2-11eb-89a8-f93ebb5bf8d7.png">

Reply:
Thanks to your feedback, i was able to use the event emitter pattern instead of the used hacky method.